src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -2122,6 +2134,9 @@ - (nullable MTRDeviceClusterData *)_clusterDataForPath:(MTRClusterPath *)cluster
     MTRDeviceClusterData * data = [_deviceController.controllerDataStore getStoredClusterDataForNodeID:_nodeID endpointID:clusterPath.endpoint clusterID:clusterPath.cluster];
     if (data != nil) {
         [_persistedClusterData setObject:data forKey:clusterPath];
+    } else {
+        // If clusterPath is in _persistedClusters and the data store returns nil for it, then the in-memory cache is now not dependable, and subscription should be reset and reestablished to reload cache from device
+        [self _resetSubscriptionWithReasonString:[NSString stringWithFormat:@"Data store has no data for cluster %@", clusterPath]];
```

Comment:
Here's a question.  Say we do that.  Then we start getting data reports from the server. That will call `_getAttributesToReportWithReportedValues:`, which will do `_cachedAttributeValueForPath`, which will land right back in this code trying to get the existing value, and nuke the subscription again, etc, no?

Actually, it's worse: just trying to start the subscription will call `_getCachedDataVersions` which will already do bad things, I expect.

At the very least, we need to clear the relevant path from `_persistedClusters` here.  But we might also need to clear other things; needs some careful thought about which ones (clusterDataToPersist?  Storage?  Others?)

Reply:
I have changed this part to reconcile `_persistedClusters` with what's in storage when this condition is detected, so the subscription reset doesn't get triggered repeatedly from the same storage data loss.